### PR TITLE
Mentors

### PR DESCRIPTION
**Pull request by Super3222**
Refer to issue https://github.com/yogstation13/yogstation/issues/1810.
I think it's worth mentioning that i almost ported BUTTBOTS while rebasing.

#### Intent of your Pull Request

From what I understand Mentors are a lower-class admin that are coded entirely differently.
Mentors are appointed by the council. They work off of a notepad file.
Mentors also have mentorSAY, for chatting like little girls, and SHOW mentor memos (those are set by admins).
Mentors also have the ability to follow players. This'll set their entire perspective on that player.

Players will have the ability to use mentorhelps in order to communicate with mentors. Mentors will also have the ability to PM players, and communicate back.

And so they get the recognition they deserve their OOC color is also different. I might as well throw in an OOC tag.

Thanks goes out to the hippiestation team.

Ports:
- https://github.com/HippieStation/HippieStation13/pull/22
- https://github.com/HippieStation/HippieStation13/pull/73
- https://github.com/HippieStation/HippieStation13/pull/170

##### Changelog

:cl:
rscadd: Ported Mentors from HippieStation.
/:cl:

Progress Bar:
Never Tested |-------------------o--| Completely Tested

### CODE REVIEW

**Comment by drderp3635** [comment]
Noice

**Comment by Jc1469** [comment]
cool. someone actually adding it.  Now make it use the ticket system :laughing: :gun:

**Comment by GrayRachnid** [comment]
How does appointing mentors work? Is it a round-by-round basis? Also, does the council know about how it works and did they make any suggestions?

**Comment by Super3222** [comment]
> How does appointing mentors work? Is it a round-by-round basis? Also, does the council know about how it works and did they make any suggestions?

Turns out I was a bit wrong. It works off of a database OR council can turn on the legacy system and use it through a notepad file.

**Comment by ShadowDeath6** [comment]
Port buttbots wtf

File: SQL/mentor.sql [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE `mentor` (
```
Comment (oisin100) [inline SQL/mentor.sql:1]: Why is this in an individual file and not in the normal schema file?
We usually prefix all our table names with "erro_" too, could you please do that?

File: SQL/mentor.sql [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  `ckey` text NOT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=latin1;
+
+CREATE TABLE `mentor_memo` (
```
Comment (oisin100) [inline SQL/mentor.sql:5]: Same goes for the prefixing here.

File: code/modules/client/verbs/who.dm [hunk 3]
```diff
@@ -2,7 +2,7 @@
 	set name = "Who"
 	set category = "OOC"
 
-	var/msg = "<b>Current Players:</b>\n"
```
Comment (oisin100) [inline code/modules/client/verbs/who.dm:5]: Why remove this?

**Review by oisin100 — CHANGES_REQUESTED** [review]

**Comment by GrayRachnid** [comment]
This merged when?

**Comment by ShadowDeath6** [comment]
see https://github.com/yogstation13/yogstation/pull/1854#pullrequestreview-26449087

**Comment by Super3222** [comment]
Oisin is taking control of this PR.

**Comment by Mrcreepssss** [comment]
hm, as far as i know mentors are like inbetween admin and player, as in no superpowers. but they do know alot about how the game works and are supposed to help newbies when asked. i personally like to do that occasionally anyway so if/when this becomes a thing i'll be a volunteer.

i'm in for this.

**Pull request closed without merging**